src/server/middleware/auth.ts

```diff
@@ -305,6 +314,11 @@ const handleKeypairAuth = async (
       throw error;
     }
 
+    const isIpInAllowlist = await checkIpInAllowlist(req);
+    if (!isIpInAllowlist) {
+      error = "Unauthorized IP address";
+      throw error;
```

Comment:
We should update `handleKeyPairAuth` to also `return` instead of throw just to be consistent. This is a good opportunity to update this 

```
return {
        isAuthed: false,
        error: <ERROR_MESSAGE>,
      };
```

Reply:
If we want to throw, we should use `throw createCustomError()`

Reply:
> If we want to throw, we should use `throw createCustomError(<error_message>,<http_status_code>, <error_code>)`

the throw is actually within a `try` block, and upon `catching`, the value of the `error` variable is assigned to the `error` being thrown. Then this `error` is being returned: `return { isAuthed: false, error };`

The function doeesn't throw, its only within the try-catch.

Reply:
ohh, yes. I missed the try-catch. Thanks